Src/FluentAssertions/Common/MemberPath.cs

```diff
@@ -64,15 +66,22 @@ private bool IsParentOf(MemberPath candidate)
             string[] candidateSegments = candidate.Segments;
 
             return candidateSegments.Length > Segments.Length &&
-                   candidateSegments.Take(Segments.Length).SequenceEqual(Segments);
+                   candidateSegments.Take(Segments.Length).SequenceEqual(Segments, MemberPathSegmentEqualityComparer);
         }
 
         private bool IsChildOf(MemberPath candidate)
         {
             string[] candidateSegments = candidate.Segments;
 
             return candidateSegments.Length < Segments.Length
-                   && candidateSegments.SequenceEqual(Segments.Take(candidateSegments.Length));
+                   && candidateSegments.SequenceEqual(Segments.Take(candidateSegments.Length),
+                       MemberPathSegmentEqualityComparer);
+        }
+
+        public MemberPath Join(MemberPath nextPath, string separator)
```

Comment:
🔧 Since you only use this member to join two paths using `[]`, I think we need to make it more explicit, e.g. something like `AsParentCollectionOf`

Reply:
done 👍